src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -64,8 +64,8 @@ protected InputStreamReader fetchGetResult(URL url) throws IOException {
     protected URL generateQueryProjectsUrl(Map<String, String> parameters) throws MalformedURLException,
             URISyntaxException {
         URIBuilder uri = new URIBuilder()
-                .setScheme(HTTPS)
-                .setHost(CRISTIN_API_HOST)
+            .setScheme(HTTPS)
+            .setHost(environment.readEnv(CRISTIN_API_HOST_ENV))
```

Comment:
Can we read the environment variable in the constructor instead to fail fast if config is wrong or missing?

Reply:
Should I pass the whole environment as parameter to constructor or only the environment variables i want to use?

Reply:
I prefer the second one. I don't think the environment reader should leak into the client.